### PR TITLE
LRUCache + ELUtils

### PR DESCRIPTION
**Pull request by pizzi80**
+ LRUCache implementation adapted to the Util codebase of Faces
+ ELUtils use the new LRUCache instead of a custom local implementation

The LRUCache can also be used in other part of the code that now use
a `Collection.synchronizeMap` over the `LRUMap`

### CODE REVIEW

File: impl/src/main/java/com/sun/faces/el/ELUtils.java [hunk 1]
```diff
@@ -252,31 +244,29 @@ public static Object coerce(Object value, Class<?> toType) {
...
+
+
+
+
+
+
+
+
+
+
```
Comment (BalusC) [inline impl/src/main/java/com/sun/faces/el/ELUtils.java:269]: What? :)

**Comment by BalusC** [comment]
Looks great :)

File: impl/src/main/java/com/sun/faces/el/ELUtils.java (hunk 1 above)
Comment (pizzi80) [reply impl/src/main/java/com/sun/faces/el/ELUtils.java]: :rofl:

**Comment by pizzi80** [comment]
I've further improved and unified with the `Cache.Factory` inside `util.Cache`

what do you think?

**Comment by BalusC** [comment]
This is a great improvement.

**Pull request merged**